src/google/protobuf/util/internal/default_value_objectwriter.cc

```diff
@@ -165,7 +165,10 @@ DefaultValueObjectWriter* DefaultValueObjectWriter::RenderBytes(
   if (current_ == NULL) {
     ow_->RenderBytes(name, value);
   } else {
-    RenderDataPiece(name, DataPiece(value, false, true));
+    // Since StringPiece is essentially a pointer, takes a copy of "value" to
+    // avoid ownership issues.
+    string_values_.push_back(new string(value.ToString()));
+    RenderDataPiece(name, DataPiece(*string_values_.back(), false, true));
```

Comment:
@xfxyjwf 

In default_value_objectwriter.cc , there is 2 step to generate json.
1. collect all necessary information into DefaultValueObjectWriter::root_
2. write json in one breath with infomation in DefaultValueObjectWriter::root_

In the first step, RenderDataPiece will be called, it just cache the StringPiece value into "root_" member(cause ownership issus here), and will not write direct to output.
In the second step, the actual write happens, but bytes field information in "root_" is "dead" already. So json will not be correct in bytes field .

Please refer to L150-L160 in default_value_objectwriter.cc, RenderString implement.